src/nvim/tui/tui.c

```diff
@@ -204,7 +204,19 @@ static void terminfo_start(UI *ui)
   data->out_isatty = os_isatty(data->out_fd);
 
   // Set up unibilium/terminfo.
+#ifdef WIN32
+  char *term = os_getenv("TERM");
+  if (term == NULL) {
+    int tty_type = uv_guess_tty(data->out_fd);
+    if (tty_type & UV_TTY_VTP || tty_type & UV_TTY_CONEMU) {
+      term = "xterm-256color";
+    } else {
+      term = "cygwin";
+    }
+  }
+#else
   const char *term = os_getenv("TERM");
```

Comment:
I think the const doesn't prevent reassigning the pointer, so this could be outside the ifdef.